src/GafferScene/RenderManifest.cpp

```diff
@@ -0,0 +1,445 @@
+					);
+				}
+			}
+		}
+	}
+
+	if( sideCarManifestPath == "" && !cryptoManifestStringData )
+	{
+		throw IECore::Exception( "No gaffer:idManifestFilePath metadata or cryptomatte metadata found." );
+	}
+
+	if( cryptoManifestStringData )
+	{
+		// Not sure it's worth doing a full text compare, but if the pointers compare equal, that's a quick
+		// and conservative test that the manifest stored in the metadata hasn't changed.
+		if( cryptoManifestStringData == m_loadedFromCryptomatteJSONStringData )
+		{
+			return;
+		}
+
+		const std::string &cryptoManifestString = cryptoManifestStringData->readable();
+		boost::iostreams::stream<boost::iostreams::array_source> stream( cryptoManifestString.c_str(), cryptoManifestString.size() );
+
+		loadCryptomatteJSON( stream );
+
+		m_loadedFromFilePath = "";
+		m_loadedFromCryptomatteJSONStringData = cryptoManifestStringData;
+
+		return;
+
+	}
+
+	std::filesystem::file_time_type currentModTime;
+	try
+	{
+		currentModTime = std::filesystem::last_write_time( sideCarManifestPath );
+	}
+	catch( std::exception &e )
+	{
+		throw IECore::Exception( std::string( "Could not find manifest file : " ) + sideCarManifestPath + " : " + e.what() );
+	}
+
+	if( sideCarManifestPath == m_loadedFromFilePath && currentModTime == m_loadedFromFilePathModTime )
+	{
+		// We're using a manifest file, and the timestamp hasn't changed since we were last loaded from this file
+		return;
+	}
+
+	if( !isCryptomatte )
+	{
+		loadEXRManifest( sideCarManifestPath.c_str() );
+	}
+	else
+	{
+		std::ifstream stream( sideCarManifestPath.c_str() );
+		loadCryptomatteJSON( stream );
+	}
+
+	m_loadedFromFilePath = sideCarManifestPath;
+	m_loadedFromFilePathModTime = currentModTime;
+	m_loadedFromCryptomatteJSONStringData.reset();
+}
+
+void RenderManifest::loadEXRManifest( const std::filesystem::path &filePath )
+{
+	m_map.clear();
+	modified();
+
+	Imf::MultiPartInputFile exrInput( filePath.generic_string().c_str() );
+	int idManifestPart = -1;
+	for( int part = 0; part < exrInput.parts (); part++ )
+	{
+		if( Imf::hasIDManifest( exrInput.header( part ) ) )
+		{
+			idManifestPart = part;
+			break;
+		}
+	}
+
+	if( idManifestPart == -1 )
+	{
+		throw IECore::Exception( "No manifest found" );
+	}
+
+	const Imf::Header header = exrInput.header( idManifestPart );
+
+	const Imf::CompressedIDManifest& compressedManifest = Imf::idManifest( header );
+	const Imf::IDManifest manifest( compressedManifest );
+	const Imf::IDManifest::ChannelGroupManifest &idManifest = manifest[0];
+
+	for( Imf::IDManifest::ChannelGroupManifest::ConstIterator i = idManifest.begin(); i != idManifest.end(); ++i )
+	{
+		m_map.insert( PathAndID( ScenePlug::stringToPath( i.text()[0] ), i.id() ) );
+	}
+}
+
+void RenderManifest::writeEXRManifest( const std::filesystem::path &filePath ) const
+{
+	if( !boost::ends_with( filePath.generic_string(), ".exr" ) )
+	{
+		throw IECore::Exception( "ID manifest file path does not end in \".exr\": " + filePath.generic_string() );
+	}
```

Comment:
Is this really our business? I think it would be the first place we checked such a thing. It might also upset some upper-case-loving folks.